### DISCUSSION TITLE
How to manage render code exceptions when debugging

### DISCUSSION THREAD

**Discussion by sbsrouteur** [QuestPDF/QuestPDF#984] [Q&A]
Hi, 

I am starting with QuestPDF atm, and I have an issue with debugging.

If a runtime exception in my code occurs during the render phase with the previewer, the exception is swallowed in the previewer and I get a white page with nothing.

It seems looking at the code that there is an exception handler in the previewer to show the exception information, but it does not seem to work for me. Or maybe I do not have properly set questpdf for that.

Is there a suggested way of handling this?

Regards,

**Comment by MarcinZiabek** [comment]
Could you please try the recently released QuestPDF Companion App? https://www.questpdf.com/companion/usage.html

**Comment by sbsrouteur** [comment]
Hi, 

I have installed the companion, but I do not have the .showcomapnion method in the document class.
I am using the questpdf in VS2022 referenced with nuget. I have installed version 2024.7.3.

The app is running as .Net7.0 console app.

**Reply by MarcinZiabek** [reply]
Please use the `2024.10.0-rc1` version of the nuget package. The feature is still in preview yet is quite capable and already stable. It may also require using the `2024.10.1` Companion App version.

**Comment by sbsrouteur** [comment]
I am now using rc4.
The exception reporting in the companion is nice, however it does not seem to support hotreload.
![image](https://github.com/user-attachments/assets/e7adc52f-525c-4453-80d4-db3221aa6c5a)

**Reply by MarcinZiabek** [reply]
Thank you for noticing this issue! 😄 

Could you please update your environment to the latest version of both nuget and Companion App?

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
